src/Uno.UI/UI/Xaml/DependencyObjectStore.cs

```diff
@@ -1252,6 +1255,210 @@ private void CleanupInheritedProperties()
 			}
 		}
 
+		public DependencyProperty? GetCorrespondingInheritedProperty(DependencyProperty property)
+		{
+			if (property.OwnerType == _originalObjectType)
+			{
+				// Uno-specific fast path.
+				return property;
+			}
+
+			if (property == RichTextBlock.CharacterSpacingProperty ||
+				property == TextBlock.CharacterSpacingProperty ||
+				property == Control.CharacterSpacingProperty ||
+				property == TextElement.CharacterSpacingProperty)
+			{
+				if (_originalObjectType == typeof(RichTextBlock)) { return RichTextBlock.CharacterSpacingProperty; }
+				else if (_originalObjectType == typeof(TextBlock)) { return TextBlock.CharacterSpacingProperty; }
```

Comment:
This is ported from WinUI and assumes that TextBlock is sealed. We should adjust to handle ImplicitTextBlock as well as Uno-specific check